test/unit-test/cellular_3gpp_api_utest.c

```diff
@@ -0,0 +1,5068 @@
+        pktStatus = atReq.respCallback( &context, &atResp, NULL, 0 );
+    }
+    else if( cbCondition == 3 )
+    {
+        char pLine[] = CELLULAR_SAMPLE_PREFIX_STRING_INPUT;
+
+        _saveData( pLine, &atResp, strlen( pLine ) + 1 );
+        pktStatus = atReq.respCallback( &context, &atResp, &cellularEidrxSettingsList, sizeof( CELLULAR_EDRX_LIST_MAX_SIZE ) );
+    }
+    else if( cbCondition == 4 )
+    {
+        char pLine[] = "+CEDRXS: 0";
+
+        _saveData( pLine, &atResp, sizeof( pLine ) );
+        pktStatus = atReq.respCallback( &context, &atResp, &cellularEidrxSettingsList, sizeof( CELLULAR_EDRX_LIST_MAX_SIZE ) );
+    }
+    else if( cbCondition == 5 )
+    {
+        char pLine = '\0';
+
+        _saveData( &pLine, &atResp, sizeof( pLine ) );
+        pktStatus = atReq.respCallback( &context, &atResp, &cellularEidrxSettingsList, sizeof( CELLULAR_EDRX_LIST_MAX_SIZE ) );
+    }
+
+    return pktStatus;
+}
+
+CellularPktStatus_t Mock_AtcmdRequestWithCallback__Cellular_RecvFuncGetSimLockStatus( CellularContext_t * pContext,
+                                                                                      CellularAtReq_t atReq,
+                                                                                      int cmock_num_calls )
+{
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    CellularContext_t context;
+    CellularATCommandResponse_t atResp;
+    CellularSimCardLockState_t cellularSimCardLockState;
+
+    memset( &atResp, 0, sizeof( CellularATCommandResponse_t ) );
+
+    /* null context */
+    if( cbCondition == 0 )
+    {
+        pktStatus = atReq.respCallback( NULL, NULL, NULL, 0 );
+    }
+    /* null at resp */
+    else if( cbCondition == 1 )
+    {
+        pktStatus = atReq.respCallback( &context, NULL, NULL, 0 );
+    }
+    /* null data pointer */
+    else if( cbCondition == 2 )
+    {
+        char pLine[] = CELLULAR_SAMPLE_PREFIX_STRING_INPUT;
+
+        _saveData( pLine, &atResp, strlen( pLine ) + 1 );
+        pktStatus = atReq.respCallback( &context, &atResp, NULL, 0 );
+    }
+    else if( cbCondition == 3 )
+    {
+        char pLine[] = CELLULAR_SAMPLE_PREFIX_STRING_INPUT;
+
+        _saveData( pLine, &atResp, strlen( pLine ) + 1 );
+        pktStatus = atReq.respCallback( &context, &atResp, &cellularSimCardLockState, sizeof( CellularSimCardLockState_t ) );
+    }
+    else if( cbCondition == 4 )
+    {
+        char pLine[] = "+CEDRXS: 0";
+
+        _saveData( pLine, &atResp, sizeof( pLine ) );
+        pktStatus = atReq.respCallback( &context, &atResp, &cellularSimCardLockState, sizeof( CellularSimCardLockState_t ) );
+    }
+
+    return pktStatus;
+}
+
+CellularATError_t MOCK_Cellular_ATGetNextTok_CALLBACK( char ** ppString,
+                                                       char ** ppTokOutput,
+                                                       int cmock_num_calls )
+{
+    CellularATError_t atCoreStatus = CELLULAR_AT_SUCCESS;

```

Comment:
This function (and many other mocks in this file) span more than 200 lines. Would it be better to have small different functions?

Reply:
Sure.